docs/source/developer/test_suite.rst

```diff
@@ -4,32 +4,80 @@ Test Suite
 
-.. code-block:: none
+The ``tests/`` directory is in the ``csunplugged/`` directory (at the same
+level as the apps) and is structured as follows:
 
+.. code-block:: none
+      
   └── tests/
       ├── general/
-      │   ├── templatetags/
-      │   ├── urls/
-      │   ├── views/
-      │   ├── test_models.py
-      │   ├── test_urls.py
-      │   └── test_views.py
+      │   ├── management/
+      │   ├── templatetags/
+      │   ├── urls/
+      │   └── views/
       ├── resources/
-      │   ├── models/
-      │   ├── resource/
-      │   ├── urls/
-      │   ├── views/
-      │   ├── test_models.py
-      │   ├── test_urls.py
-      │   └── test_views.py
+      │   ├── loaders/
+      │   ├── models/
+      │   ├── resource/
+      │   ├── urls/
+      │   ├── views/
+      │   └── ResourcesTestDataGenerator.py
       ├── topics/
-      │   ├── models/
-      │   ├── urls/
-      │   ├── views/
-      │   ├── test_models.py
-      │   ├── test_urls.py
-      │   └── test_views.py
-      └── BaseTest.py
+      │   ├── loaders/
+      │   │   └── assets/
+      │   ├── models/
+      │   ├── urls/
+      │   ├── views/
+      │   └── TopicsTestDataGenerator.py
+      ├── utils
+      │   └── errors/
+      └── BaseTestWithDB.py
+
+Note that each app being tested has it's own folder, and this is then broken down further into the component being tested (i.e. views, urls, models, etc).
+
+Items of interest from this diagram:
+
+- ``models/`` - contains a test file for each model in the app.
+
+- ``urls/`` - contains a test file for each url in the app.
+
+- ``views/`` - contains a test file for each view in the app.
+
+- ``loaders/`` - contains a test file for each loader in the app as well as an
+  ``assets/`` directory. Test yaml files should be saved in the corresponding loader directory within ``assets/`` and should mimick the folder structure of the app where necessary.
+
+- Test Data Generators - these are classes used to generate place holder data in
+  the database. You should use the methods in these files when testing a model that contains a foreign key and/or many-to-many field that needs to be
+  populated.
+
+- ``BaseTestWithDB.py`` - this class inherits the Django ``TestCase`` class, and
+  creates and logs in a user. This is the base test class that all other tests
+  should inherit from in order to interact with the database.
+
+- ``utils/errors/`` - contains test error classes for testing exceptions are
+  raised correctly by the loaders.
+
+Adding Tests
+==============================================================================
+
+When writing a new test function, it is important that the method name is as 
+descriptive as possible. The method name should also be prefixed with ``test_``
+as the test suite will only execute methods with this prefix.
+
+.. note::
+
+  We use `Codecov`_ to check the coverage of our tests. Every Pull Request should
```

Comment:
Doesn't need to specifically increase coverage, but should cover 100% of the difference (which therefore increases coverage).